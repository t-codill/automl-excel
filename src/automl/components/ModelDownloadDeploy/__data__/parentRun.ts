import { RunHistoryAPIsModels } from "@vienna/runhistory";

export const parentRun: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDetailsDto = {
    runId: "test-run-id",
    properties: {
        DataPrepJsonString: "{\"features\":[\"timeStamp\",\"precip\",\"temp\"]}"
    }
};
