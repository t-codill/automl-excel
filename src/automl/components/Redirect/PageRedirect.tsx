import * as React from "react";
import { generatePath, Redirect } from "react-router-dom";
import { IComponentWithRoute } from "../../common/IComponentWithRoute";
import { PageRedirectLinkRender } from "./PageRedirectLinkRender";
import { PageRedirectRender } from "./PageRedirectRender";
import { RedirectLink } from "./RedirectLink";

export type RouteProp<TRouteProp> = { [K in keyof TRouteProp]: string | number | boolean };

export function PageRedirectLink<TProp extends RouteProp<TProp>>(
    linkContent: React.ReactNode,
    component: IComponentWithRoute<TProp>,
    routeProps: Readonly<TProp>
): React.ReactNode {
    return <PageRedirectLinkRender expendedRoutePath={generatePath(component.routePath, routeProps)} >
        {linkContent}
    </PageRedirectLinkRender>;
}

export function PageRedirect<TProp extends RouteProp<TProp>>(
    component: IComponentWithRoute<TProp>,
    routeProps: Readonly<TProp>,
    /**
     * pass true for not pushing history.
     */
    noPush = false
): React.ReactNode {
    return <PageRedirectRender noPush={noPush || false} expendedRoutePath={generatePath(component.routePath, routeProps)} />;
}

export function BaseRedirectLink<TProp extends RouteProp<TProp>>(
    linkContent: React.ReactNode,
    component: IComponentWithRoute<TProp>,
    routeProps: Readonly<TProp>
): React.ReactNode {
    return (
        <RedirectLink to={`${generatePath(component.routePath, routeProps)}`}>
            {linkContent}
        </RedirectLink>
    );
}

export function BaseRedirect<TProp extends RouteProp<TProp>>(
    component: IComponentWithRoute<TProp>,
    routeProps: Readonly<TProp>,
    /**
     * Pass true for not pushing history.
     */
    noPush = false
): React.ReactNode {
    if (noPush) {
        return <Redirect to={`${generatePath(component.routePath, routeProps)}`} push={false} />;
    }
    else {
        return <Redirect to={`${generatePath(component.routePath, routeProps)}`} push={true} />;
    }
}
