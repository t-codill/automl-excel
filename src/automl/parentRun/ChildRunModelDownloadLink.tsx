import { Icon, Link, Spinner, SpinnerSize } from "office-ui-fabric-react";
import * as React from "react";
import { download } from "../common/utils/download";
import { BaseComponent } from "../components/Base/BaseComponent";
import { ArtifactService, IRunDataForArtifact } from "../services/ArtifactService";

interface IChildRunModelDownloadLinkProps {
    run: IRunDataForArtifact;
}

export class ChildRunModelDownloadLink extends BaseComponent<
    IChildRunModelDownloadLinkProps,
    {
        downloading: boolean;
    },
    { artifactService: ArtifactService }
    > {
    protected readonly serviceConstructors = { artifactService: ArtifactService };
    protected getData?: (() => Promise<void>) | undefined;
    constructor(props: IChildRunModelDownloadLinkProps) {
        super(props);
        this.state = {
            downloading: false
        };
    }
    public render(): React.ReactNode {
        if (this.props.run.status !== "Completed") {
            return <></>;
        }
        if (this.state.downloading) {
            return <Spinner size={SpinnerSize.small} />;
        }

        return <Link onClick={this.onClick}>
            <Icon iconName="Download" /> Download
        </Link>;
    }

    private readonly onClick = async () => {
        this.setState({ downloading: true });
        const uri = await this.services.artifactService.getModelUrl(this.props.run);
        if (uri === undefined) {
            return;
        }
        this.setState({ downloading: false });
        if (uri === null) {
            return;
        }
        this.logUserAction("ModelDownloadFromTable", { childRunId: this.props.run.runId });
        download(uri, "model.pkl");
    }
}
