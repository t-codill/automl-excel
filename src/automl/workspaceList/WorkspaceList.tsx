import { AzureMachineLearningWorkspacesModels } from "@azure/arm-machinelearningservices";
import * as React from "react";
import { BaseContext } from "../common/context/BaseContext";
import { getResourceGroupName } from "../common/utils/getResourceGroupName";
import { nameof } from "../common/utils/nameof";
import { DataTable } from "../components/DataTable/DataTable";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { BaseRedirectLink } from "../components/Redirect/PageRedirect";
import { Main } from "../Main";
import { WorkSpaceService } from "../services/WorkSpaceService";

export interface IWorkspaceListProps {
    subscriptionId: string;
}

export class WorkspaceList extends React.Component<IWorkspaceListProps, {
    workspaces: AzureMachineLearningWorkspacesModels.Workspace[] | undefined;
}> {
    public static routePath = `/subscriptions/:${nameof<IWorkspaceListProps>("subscriptionId")}`;
    public static contextType = BaseContext;
    public context!: React.ContextType<typeof BaseContext>;
    private workSpaceService!: WorkSpaceService;
    public componentDidMount(): void {
        this.workSpaceService = new WorkSpaceService({
            ...this.props,
            ...this.context,
            resourceGroupName: "",
            workspaceName: "",
            discoverUrls: {},
            location: "",
            onError: () => { return; }
        });
        this.loadData();
    }
    public componentWillUnmount(): void {
        this.workSpaceService.dispose();
    }
    public render(): React.ReactNode {
        if (!this.state || !this.state.workspaces) {
            return <PageLoadingSpinner />;
        }

        return (
            <DataTable
                items={this.state.workspaces}
                sortColumnFieldName="name"
                columns={[
                    { field: "name", minWidth: 200, maxWidth: 500, render: this.renderWorkSpaceLink },
                    { field: "workspaceId", minWidth: 200 }
                ]}
            />
        );
    }

    private readonly renderWorkSpaceLink = (content: React.ReactNode, data: AzureMachineLearningWorkspacesModels.Workspace) => {
        if (!data.workspaceId) {
            return undefined;
        }
        return BaseRedirectLink(content, Main, {
            subscriptionId: this.props.subscriptionId,
            resourceGroupName: getResourceGroupName(data.id) || "",
            workspaceName: data.name || ""
        });
    }

    private async loadData(): Promise<void> {
        const workspaces = await this.workSpaceService.listWorkspaces();
        if (!workspaces) {
            return;
        }
        this.setState({ workspaces });
    }
}
