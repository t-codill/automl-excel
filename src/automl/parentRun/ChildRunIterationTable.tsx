import { MessageBar } from "office-ui-fabric-react";
import * as React from "react";
import { ChildRun } from "../childRun/ChildRun";
import { calculateDuration } from "../common/utils/calculateDuration";
import { generateRunName } from "../common/utils/generateRunName";
import { isRunCompleted, isTerminatedRunStatus } from "../common/utils/run";
import { safeParseJson } from "../common/utils/safeParseJson";
import { DataTable } from "../components/DataTable/DataTable";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { PageRedirectLink } from "../components/Redirect/PageRedirect";
import { ChildRunModelDownloadLink } from "./ChildRunModelDownloadLink";
import { IParentRunData } from "./ParentRun";

interface IChildRunIterationTableRow {
    runId: string | undefined;
    experimentId: string | undefined;
    status: string | undefined;
    createdUtc: Date | undefined;
    iteration: number | undefined;
    run_name: string | undefined;
    score: number;
    duration: string | undefined;
    parentRunId: string | undefined;
}

export class ChildRunIterationTable
    extends React.PureComponent<IParentRunData> {
    public readonly render = (): React.ReactNode => {
        if (!this.props || !this.props.run || !this.props.childRuns) {
            return <PageLoadingSpinner />;
        }
        if (isTerminatedRunStatus(this.props.run.status) &&
            this.props.childRuns.length < 1) {
            return (
                <MessageBar>No data for visualization</MessageBar>
            );
        }

        const ascending = this.props.run.properties && (safeParseJson(this.props.run.properties.AMLSettingsJsonString).metric_operation === "minimize");
        const convertedObj: IChildRunIterationTableRow[] = this.props.childRuns.map((run) => {
            return {
                runId: run.runId,
                experimentId: run.experimentId,
                status: run.status,
                createdUtc: run.createdUtc,
                iteration: run.properties ? parseInt(run.properties.iteration, 10) : undefined,
                run_name: generateRunName(run),
                score: isRunCompleted(run) && run.properties && run.properties.score ? Number(run.properties.score) : NaN,
                duration: calculateDuration(run.startTimeUtc, run.endTimeUtc),
                parentRunId: run.parentRunId
            };
        });
        return <DataTable
            enableShimmer={this.props.childRuns.length < 1}
            shimmerLines={5}
            items={convertedObj}
            columns={[
                { field: "iteration", header: "Iteration", minWidth: 50, maxWidth: 60 },
                { field: "run_name", header: "Name", minWidth: 100, maxWidth: 240, render: this.renderChildRunLink },
                {
                    field: "score",
                    header: (this.props.run.properties && this.props.run.properties.primary_metric) ? this.props.run.properties.primary_metric : "Primary Metric",
                    minWidth: 80,
                    maxWidth: 140,
                    nanAsMax: ascending
                },
                { field: "status", header: "Status", minWidth: 80, maxWidth: 80 },
                { field: "createdUtc", header: "Created", minWidth: 80, maxWidth: 150 },
                { field: "duration", header: "Duration", minWidth: 80, maxWidth: 80 },
                { field: "iteration", header: "Model", minWidth: 60, maxWidth: 80, render: this.renderDownload }
            ]}
            sortColumnFieldName="score"
            sortDescending={!ascending}
        />;
    }

    private readonly renderChildRunLink = (content: React.ReactNode, run: IChildRunIterationTableRow) => {
        if (!run.runId || !run.experimentId) {
            return undefined;
        }
        return PageRedirectLink(content, ChildRun, { runId: run.runId, experimentName: this.props.experimentName });
    }
    private readonly renderDownload = (_content: React.ReactNode, run: IChildRunIterationTableRow) => {
        return <ChildRunModelDownloadLink run={run} />;
    }
}
