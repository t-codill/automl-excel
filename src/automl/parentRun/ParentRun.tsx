import { RunHistoryAPIsModels } from "@vienna/runhistory";

import { ICommandBarItemProps } from "office-ui-fabric-react";
import * as React from "react";
import { IDictionary } from "../common/IDictionary";
import { PageNames } from "../common/PageNames";
import { isAutoRefreshStatus } from "../common/utils/isAutoRefreshStatus";
import { nameof } from "../common/utils/nameof";
import { BasePage } from "../components/Base/BasePage";
import { ConfirmationDialog } from "../components/Dialog/ConfirmationDialog";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { PopupProgressIndicator } from "../components/Progress/PopupProgressIndicator";
import { PageRedirect } from "../components/Redirect/PageRedirect";
import { LogDetails } from "../logDetails/LogDetails";
import { RunList } from "../runList/RunList";
import { ArtifactService } from "../services/ArtifactService";
import { JasmineService } from "../services/JasmineService";
import { RunHistoryService, RunMetricType } from "../services/RunHistoryService";
import { ParentRunGrid } from "./ParentRunGrid";
import { RunStatus } from "./RunStatus";

export interface IParentRunRouteProps {
    experimentName: string;
    runId: string;
}

export interface IParentRunData {
    run: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDetailsDto | undefined;
    childRuns: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDto[] | undefined;
    childRunMetrics: Array<IDictionary<RunMetricType>> | undefined;
    experimentName: string;
}

export interface IParentRunState extends IParentRunData {
    hideCancelDialog: boolean;
    canceling: boolean;
    goToLog: boolean;
}

export class ParentRun extends BasePage<
    IParentRunRouteProps,
    IParentRunState,
    { runHistoryService: RunHistoryService; artifactService: ArtifactService; jasmineService: JasmineService }> {
    public static routePath = `experiments/:${nameof<IParentRunRouteProps>("experimentName")}/parentrun/:${nameof<IParentRunRouteProps>("runId")}`;
    protected readonly header = "Run Detail";
    protected readonly pageName = PageNames.ParentRun;
    protected readonly serviceConstructors = { runHistoryService: RunHistoryService, artifactService: ArtifactService, jasmineService: JasmineService };
    protected autoRefreshInMs = 30000;
    protected readonly navigationBarButtons: ICommandBarItemProps[] = [
        {
            key: "log",
            text: "Logs",
            iconProps: {
                iconName: "TextDocument"
            },
            onClick: () => {
                this.logUserAction("Log", { parentRunId: this.props.runId });
                this.goToLog();
            }
        },
        {
            key: "cancelRun",
            text: "Cancel Run",
            iconProps: {
                iconName: "Cancel"
            },
            onClick: () => {
                this.logUserAction("CancelRun", { parentRunId: this.props.runId });
                this.showDialog();
            }
        }
    ];

    public constructor(props: IParentRunRouteProps) {
        super(props);
        this.state = {
            run: undefined,
            childRuns: undefined,
            childRunMetrics: undefined,
            experimentName: props.experimentName,
            hideCancelDialog: true,
            canceling: false,
            goBack: false,
            goToLog: false
        };
    }

    public readonly render = (): React.ReactNode => {
        if (this.state.goToLog) {
            return PageRedirect(LogDetails, {
                runId: this.props.runId,
                runType: "parent",
                experimentName: this.props.experimentName,
            });
        }
        if (this.state.goBack) {
            return PageRedirect(RunList, {});
        }
        if (!this.state || !this.state.run) {
            return <PageLoadingSpinner />;
        }
        const run = this.state.run;
        const parentRunGridProps = {
            ...this.state,
            onModelRegister: this.refresh.bind(this)
        };
        return <>
            <RunStatus run={run} experimentName={this.props.experimentName} />
            <ParentRunGrid {...parentRunGridProps} />
            {this.state.canceling &&
                <PopupProgressIndicator
                    title="Canceling"
                    description={`Canceling run: ${this.state.run.runId}`}
                />
            }
            <ConfirmationDialog
                title="Cancel Run?"
                subText={`Are you sure you want to cancel run: ${this.state.run.runId}`}
                hidden={this.state.hideCancelDialog}
                onConfirm={this.cancelRun}
                onClose={this.closeDialog}
            />
        </>;
    }

    protected readonly getData = async () => {
        await this.getRunDetail();
        await this.getChildren();
    }

    private readonly goToLog = async () => {
        this.setState({ goToLog: true });
    }

    private readonly getChildren = async () => {
        if (!this.state || !this.state.run || !this.state.run.runId) {
            return;
        }

        // if run has a parent run, run is a child run
        if (this.state.run.parentRunId) {
            return;
        }

        const childRuns = await this.services.runHistoryService.getChildRuns(this.props.runId, this.props.experimentName);
        if (!childRuns) {
            return;
        }

        this.setState({ childRuns });
        if (!this.state.run.properties) {
            this.setState({ childRunMetrics: [] });
            return;
        }
        const childRunMetrics = await this.services.runHistoryService.getChildRunMetrics(childRuns, [this.state.run.properties.primary_metric], this.props.experimentName);
        if (!childRunMetrics) {
            return;
        }

        this.setState({ childRunMetrics });
    }

    private readonly getRunDetail = async () => {
        const run = await this.services.runHistoryService.getRun(this.props.runId, this.props.experimentName);
        if (!run || !run.runId) {
            return;
        }
        this.setState({
            run,
            experimentName: this.props.experimentName
        });
        if (this.autoRefreshEnabled === undefined) {
            this.changeAutoRefresh(isAutoRefreshStatus(run.status));
        }
        const cancelButton = this.navigationBarButtons.find((button) => {
            return button.key === "cancelRun";
        });
        if (cancelButton) {
            cancelButton.disabled = !isAutoRefreshStatus(this.state.run && this.state.run.status) || run.target === "local";
        }
        this.refreshButtons();
    }

    private readonly showDialog = (): void => {
        this.setState({ hideCancelDialog: false });
    }

    private readonly closeDialog = (): void => {
        this.setState({ hideCancelDialog: true });
    }

    private readonly cancelRun = async () => {
        this.closeDialog();

        if (!this.state.run || !this.state.run.runId || !this.state.experimentName) {
            return;
        }
        this.setState({ canceling: true });
        const response = await this.services.jasmineService.cancelRun(this.state.run.runId, this.state.experimentName);
        this.setState({ canceling: false });
        if (response) {
            await this.refresh();
        }
    }
}
