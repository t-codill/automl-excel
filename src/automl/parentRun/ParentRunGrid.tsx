import * as React from "react";
import { shouldShowParentRunModel } from "../common/utils/run";
import { GridItem } from "../components/Grid/GridItem";
import { ChildRunIterationTable } from "./ChildRunIterationTable";
import { IParentRunData } from "./ParentRun";
import { ParentRunModel } from "./ParentRunModel";
import { ParentRunSkylineChart } from "./ParentRunSkylineChart";
import { ParentRunSummary } from "./ParentRunSummary";
import { ParentSettings } from "./ParentSettings";

export interface IParentRunGridProps extends IParentRunData {
    onModelDeploy(): void;
}
export class ParentRunGrid extends React.Component<IParentRunGridProps> {
    public readonly render = (): React.ReactNode => {
        const modelVisible = shouldShowParentRunModel(this.props.run && this.props.run.status);
        return (
            <div className="ms-Grid" dir="ltr">
                <div className="ms-Grid-col ms-sm12 ms-md12 ms-lg12 ms-xl9">
                    <GridItem title="Iteration Chart">
                        <ParentRunSkylineChart  {...this.props} />
                    </GridItem>
                    <GridItem title="Iterations">
                        <ChildRunIterationTable  {...this.props} />
                    </GridItem>
                </div>
                <div className="ms-Grid-col ms-sm12 ms-md12 ms-lg12 ms-xl3">
                    {modelVisible && <ParentRunModel {...this.props} />}
                    <GridItem title="Run Summary">
                        <ParentRunSummary run={this.props.run} />
                    </GridItem>
                    <GridItem title="Run Settings">
                        <ParentSettings run={this.props.run} />
                    </GridItem>
                </div>
            </div>
        );
    }
}
