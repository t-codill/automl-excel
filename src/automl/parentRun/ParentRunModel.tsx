import { RunHistoryAPIsModels } from "@vienna/runhistory";
import { isEqual, orderBy } from "lodash";
import { Icon, Link } from "office-ui-fabric-react";
import * as React from "react";
import { PageNames } from "../common/PageNames";
import { generateRunName } from "../common/utils/generateRunName";
import { safeParseJson } from "../common/utils/safeParseJson";
import { BaseComponent } from "../components/Base/BaseComponent";
import { ModelDeploy } from "../components/ModelDownloadDeploy/ModelDeploy";
import { ModelDownload } from "../components/ModelDownloadDeploy/ModelDownload";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { ArtifactService } from "../services/ArtifactService";
import { IParentRunData } from "./ParentRun";
import { IParentRunGridProps } from "./ParentRunGrid";

export interface IParentRunModelState {
    modelName: string | undefined;
    uri: string | undefined;
    bestRun: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDto | undefined;
}

export class ParentRunModel extends BaseComponent<IParentRunGridProps, IParentRunModelState, { artifactService: ArtifactService }> {

    protected serviceConstructors = { artifactService: ArtifactService };
    constructor(props: IParentRunGridProps) {
        super(props);
        this.state = {
            modelName: undefined,
            uri: undefined,
            bestRun: undefined
        };
    }

    public async componentDidUpdate(prevProps: IParentRunData): Promise<void> {
        if (isEqual(this.props.childRuns, prevProps.childRuns)) {
            return;
        }
        await this.refresh();
    }

    public readonly render = (): React.ReactNode => {
        if (!this.props || !this.props.run || !this.props.childRuns) {
            return <PageLoadingSpinner />;
        }

        return <>
            <div style={{
                display: "flex",
                justifyContent: "space-between"
            }}>
                <ModelDownload
                    pageName={PageNames.ParentRun}
                    experimentName={this.props.experimentName}
                    run={this.state.bestRun}
                    modelName={this.state.modelName}
                    modelUri={this.state.uri} />
                <ModelDeploy
                    pageName={PageNames.ParentRun}
                    experimentName={this.props.experimentName}
                    run={this.state.bestRun}
                    parentRun={this.props.run}
                    modelName={this.state.modelName}
                    modelUri={this.state.uri}
                    modelId={this.state.bestRun && this.state.bestRun.tags && this.state.bestRun.tags.model_id}
                    onModelRegister={this.props.onModelRegister} />
            </div>
            <div style={{
                textAlign: "right",
                marginBottom: "1em"
            }}>
                <Link target="_blank" href="https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-create-portal-experiments#deploy-model">
                    Learn more about deploying models <Icon iconName="NavigateExternalInline" />
                </Link>
            </div>
        </>;
    }
    protected readonly getData = async () => {
        if (!this.props.childRuns || !this.props.run || !this.props.run.properties) {
            this.setState({
                bestRun: undefined,
                modelName: undefined,
                uri: undefined
            });
            return;
        }
        const min = safeParseJson(this.props.run.properties.AMLSettingsJsonString).metric_operation === "minimize";
        const scoredRuns = this.props.childRuns.map((r) => ({
            ...r,
            runId: r.runId,
            parentRunId: r.parentRunId,
            status: r.status,
            score: r.properties && parseFloat(r.properties.score)
        }));

        const sortedRuns = orderBy(scoredRuns, ["score"], [min ? "asc" : "desc"]);
        for (const bestRun of sortedRuns) {
            if (bestRun.score === undefined || isNaN(bestRun.score)) {
                continue;
            }
            const uri = await this.services.artifactService.getModelUrl({
                runId: bestRun.runId,
                parentRunId: bestRun.parentRunId,
                status: bestRun.status
            });
            if (!uri) {
                continue;
            }
            this.setState({
                bestRun,
                modelName: generateRunName(bestRun),
                uri
            });
            return;
        }
        this.setState({
            bestRun: undefined,
            modelName: undefined,
            uri: undefined
        });
    }
}
