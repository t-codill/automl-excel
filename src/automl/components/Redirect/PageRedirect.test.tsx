import { shallow } from "enzyme";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ParentRun } from "../../parentRun/ParentRun";
import { WorkspaceList } from "../../workspaceList/WorkspaceList";
import { BaseRedirect, BaseRedirectLink, PageRedirect, PageRedirectLink } from "./PageRedirect";

describe("PageRedirect", () => {
    it("should render PageRedirectLink", () => {
        const tree = shallow(
            <BrowserRouter>
                {PageRedirectLink("Parent Run", ParentRun, { experimentName: "expName", runId: "AutoML_123" })}
            </BrowserRouter>);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render PageRedirect", () => {
        const tree = shallow(
            <BrowserRouter>
                {PageRedirect(ParentRun, { experimentName: "expName", runId: "AutoML_123" })}
            </BrowserRouter>);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render BaseRedirectLink", () => {
        const tree = shallow(
            <BrowserRouter>
                {BaseRedirectLink("WorkspaceList", WorkspaceList, { subscriptionId: "00000000-0000-0000-0000-000000000000" })}
            </BrowserRouter>);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render BaseRedirect", () => {
        const tree = shallow(
            <BrowserRouter>
                {BaseRedirect(WorkspaceList, { subscriptionId: "00000000-0000-0000-0000-000000000000" })}
            </BrowserRouter>);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render BaseRedirect with no push", () => {
        const tree = shallow(
            <BrowserRouter>
                {BaseRedirect(WorkspaceList, { subscriptionId: "00000000-0000-0000-0000-000000000000" }, true)}
            </BrowserRouter>);
        expect(tree)
            .toMatchSnapshot();
    });
});
