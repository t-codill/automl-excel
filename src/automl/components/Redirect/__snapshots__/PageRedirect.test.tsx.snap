// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageRedirect should render BaseRedirect 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Redirect
    push={true}
    to="/subscriptions/00000000-0000-0000-0000-000000000000"
  />
</Router>
`;

exports[`PageRedirect should render BaseRedirect with no push 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Redirect
    push={false}
    to="/subscriptions/00000000-0000-0000-0000-000000000000"
  />
</Router>
`;

exports[`PageRedirect should render BaseRedirectLink 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <RedirectLink
    to="/subscriptions/00000000-0000-0000-0000-000000000000"
  >
    WorkspaceList
  </RedirectLink>
</Router>
`;

exports[`PageRedirect should render PageRedirect 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PageRedirectRender
    expendedRoutePath="experiments/expName/parentrun/AutoML_123"
    noPush={false}
  />
</Router>
`;

exports[`PageRedirect should render PageRedirectLink 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PageRedirectLinkRender
    expendedRoutePath="experiments/expName/parentrun/AutoML_123"
  >
    Parent Run
  </PageRedirectLinkRender>
</Router>
`;
